src/main/java/com/saarthak/healthcheck/CouchBaseHealthCheck.java

```diff
@@ -26,28 +26,24 @@ public CouchBaseHealthCheck(final CouchbaseClient couchbaseClient) {
     @Override
     protected Result check() throws Exception {
         log.info("Running health check for couch base");
-
-        JsonArray nodes = couchbaseClient.getNodes();
-        Boolean isHealthy = true;
-        final List<Boolean> nodesList = new ArrayList<>();
-        List<String> healthyNodes = new ArrayList<>();
-        List<String> unHealthyNodes = new ArrayList<>();
-        nodes.iterator().forEachRemaining(node -> {
-            JsonObject obj = (JsonObject) node;
-            boolean isNodeHealthy = "healthy".equals(obj.get("status"));
-            nodesList.add(isNodeHealthy);
+        final JsonArray nodes = couchbaseClient.getNodes();
+        final List<String> healthyNodes = new ArrayList<>();
+        final List<String> unhealthyNodes = new ArrayList<>();
+        nodes.forEach(node -> {
+            final JsonObject obj = (JsonObject) node;
+            final boolean isNodeHealthy = "healthy".equals(obj.get("status"));
             if (isNodeHealthy) {
                 healthyNodes.add(obj.getString("hostname"));
             } else {
-                unHealthyNodes.add(obj.getString("hostname"));
+                unhealthyNodes.add(obj.getString("hostname"));
             }
         });
 
-        for (Boolean node : nodesList) {
-            isHealthy = isHealthy && node;
-        }
+        //isHealthy is derived to be true when 50% or more nodes are healthy.
+        final boolean isHealthy = healthyNodes.size() >= unhealthyNodes.size();
```

Comment:
We should also take in a minimumQuorum from the user to allow him to specify his own percentage of healthy nodes that can classify the cluster as healthy.
For example, someone might consider the cluster as Healthy even if 1 of the nodes out of n nodes is healthy.